lib/class-wp-theme-json-resolver.php

```diff
@@ -71,6 +71,85 @@ private static function get_from_file( $file_path ) {
 		return $config;
 	}
 
+	/**
+	 * Processes a tree from i18n-theme.json into a linear array
+	 * containing the a translatable path from theme.json and an array
+	 * of properties that are translatable.
+	 *
+	 * @param array $file_structure_partial A part of a theme.json i18n tree.
+	 * @param array $current_path           An array with a path on the theme.json i18n tree.
+	 *
+	 * @return array An array of arrays each one containing a translatable path and an array of properties that are translatable.
+	 */
+	private static function theme_json_i18_file_structure_to_paths( $file_structure_partial, $current_path = array() ) {
```

Comment:
This processing logic is tied to how presets are structured and can't be used for anything else. I don't anticipate that we want to translate any other thing in theme.json either. So, given this context, what do you think of making the naming of everything this PR adds more explicit: `translate_presets` instead of `apply_theme_json_translations`, `get_presets_to_translate` instead of `get_theme_json_i18n`, etc.

Reply:
Ideally, it isn't tied to `themes.json` as well. We need it for `block.json` 😄

Reply:
oh, so is it the idea that the file `lib/experimental-i18n-theme.json` and this code is used for both block.json and theme.json? If that's the case, it sounds like something independent from the `WP_Theme_JSON_Resolver` class.

Reply:
The definition with config can (or maybe even should) be in their own files for blocks and themes. However the logic that applies translation functions could be shared 😃

Reply:
Well, it is a fact that this code and even the structure is focused on presets.
All these functions were not implemented with block.json in consideration, but I guess we can in fact have a genetic data structure translation tool outside WP_Theme_JSON_Resolver (but used by WP_Theme_JSON_Resolver) this independent class could be used by both block.json and by theme.json.

We would need a specification format that can handle block.json and theme.json independent of  presets I guess something like that would work
```
{
	'*': {
		"settings": {
			"typography": {
				"fontSizes": {
					'*': [ 
						'name'
					]
				}
			}
		}
	}
}
```

But this is not very future proof as soon as we had nesting this specification would not work.

Being totally generic seems to bring more complexity to the specification structure. Maybe it is fine to have a set of functions focused on theme.json and presets and then we have something similar for block.json?

Reply:
It can be two different helpers if that makes it simpler 👍

Reply:
Hi @nosolosw, I renamed the functions as suggested.